### PR TITLE
Use heightmap to get correct groundlevel, use AircraftInfo for correct ground level

### PR DESCRIPTION
**Pull request by teees**
Same as in #10162 for the AttackPlane and ReturnOnIdle traits.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Air/AttackPlane.cs [hunk 1]
```diff
@@ -40,7 +40,9 @@ public override Activity GetAttackActivity(Actor self, Target newTarget, bool al
 		protected override bool CanAttack(Actor self, Target target)
 		{
 			// Don't fire while landed or when outside the map.
-			return base.CanAttack(self, target) && self.CenterPosition.Z > 0 && self.World.Map.Contains(self.Location);
+			return base.CanAttack(self, target)
+				&& self.World.Map.DistanceAboveTerrain(self.CenterPosition).Length > 0
```
Comment (atlimit8) [inline OpenRA.Mods.Common/Traits/Air/AttackPlane.cs]: `&& self.World.Map.DistanceAboveTerrain(self.CenterPosition).Length >= aircraft.MinAirborneAltitude` where `readonly AircraftInfo aircraft` and `aircraft = self.Info.TraitInfo<AircraftInfo>()` in the constructor.

File: OpenRA.Mods.Common/Traits/Air/ReturnOnIdle.cs [hunk 2]
```diff
@@ -22,7 +22,7 @@ class ReturnOnIdle : INotifyIdle
 		public void TickIdle(Actor self)
 		{
 			// We're on the ground, let's stay there.
-			if (self.CenterPosition.Z == 0)
+			if (self.World.Map.DistanceAboveTerrain(self.CenterPosition).Length == 0)
```
Comment (atlimit8) [inline OpenRA.Mods.Common/Traits/Air/ReturnOnIdle.cs]: Also, something similar here.

**Comment by teees** [comment]
changed and rebased

**Comment by abcdefg30** [comment]
Looks good to me. :+1: / :white_check_mark:

**Comment by penev92** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/713bfc4662d005131f27e4ab902c54e9f149e699)

**Pull request merged**